CRM/Utils/System/WordPress.php

```diff
@@ -926,20 +926,42 @@ public function getUniqueIdentifierFromUserObject($user) {
    * @inheritDoc
    */
   public function getLoginURL($destination = '') {
-    $config = CRM_Core_Config::singleton();
-    $loginURL = wp_login_url();
-    return $loginURL;
+    return wp_login_url($destination);
   }
 
   /**
-   * FIXME: Do something.
-   *
    * @param \CRM_Core_Form $form
    *
    * @return NULL|string
    */
   public function getLoginDestination(&$form) {
-    return NULL;
+    $args = NULL;
+
+    $id = $form->get('id');
+    if ($id) {
+      $args .= "&id=$id";
+    }
+    else {
+      $gid = $form->get('gid');
+      if ($gid) {
+        $args .= "&gid=$gid";
+      }
+      else {
+        // Setup Personal Campaign Page link uses pageId
+        $pageId = $form->get('pageId');
+        if ($pageId) {
+          $component = $form->get('component');
+          $args .= "&pageId=$pageId&component=$component&action=add";
+        }
+      }
+    }
+
+    $destination = NULL;
+    if ($args) {
+      // append destination so user is returned to form they came from after login
+      $destination = CRM_Utils_System::url(CRM_Utils_System::currentPath(), 'reset=1' . $args);
```

Comment:
Does this need to be forced to the front end or not?

Reply:
TBH this was copied from https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/System/DrupalBase.php#L488-L505  
It works as is for the case described in https://lab.civicrm.org/dev/core/-/issues/2436
I'll remove the other bits and keep it simple.  If anyone wants this for PCP's they can add later.

Reply:
Drupal doesn't have the same concern about front end v backend as WordPress thoughts @kcristiano @haystack

Reply:
I'll do an `r-run`  I am concerned about who it will get the URLs depending on front-end vs back-end

Reply:
I can't envisage a situation where this redirect would happen on the back end. If someone can give an example, then yes I'd agree that forcing front end would be necessary.